### ISSUE TITLE
test_auth_v2_migration.test_auth_v2_migration fails due to "Role user_after_migration already exists."

### ISSUE THREAD

**Issue by tchaikov** [scylladb/scylladb#18319]
```
request = <FixtureRequest for <Function test_auth_v2_migration>>
manager = <test.pylib.manager_client.ManagerClient object at 0x7f57151cf7d0>

    @pytest.mark.asyncio
    async def test_auth_v2_migration(request, manager: ManagerClient):
        # First, force the first node to start in legacy mode due to the error injection
        cfg = {'error_injections_at_startup': ['force_gossip_based_join']}
    
        servers = [await manager.server_add(config=cfg)]
        # Disable injections for the subsequent nodes - they should fall back to
        # using gossiper-based node operations
        del cfg['error_injections_at_startup']
    
        servers += [await manager.server_add(config=cfg) for _ in range(2)]
        cql = manager.cql
        assert(cql)
    
        logging.info("Waiting until driver connects to every server")
        hosts = await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
    
        logging.info("Checking the upgrade state on all nodes")
        for host in hosts:
            status = await manager.api.raft_topology_upgrade_status(host.address)
            assert status == "not_upgraded"
    
        await populate_auth_v1_data(manager)
        await warmup_v1_static_values(manager, hosts)
    
        logging.info("Triggering upgrade to raft topology")
        await manager.api.upgrade_to_raft_topology(hosts[0].address)
    
        logging.info("Waiting until upgrade finishes")
        await asyncio.gather(*(wait_until_topology_upgrade_finishes(manager, h.address, time.time() + 60) for h in hosts))
    
        logging.info("Checking migrated data in system_auth_v2")
        await check_auth_v2_data_migration(manager, hosts)
    
        logging.info("Checking auth statements after migration")
>       await check_auth_v2_works(manager, hosts)

test/auth_cluster/test_auth_v2_migration.py:165: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

manager = <test.pylib.manager_client.ManagerClient object at 0x7f57151cf7d0>
hosts = [<Host: 127.101.142.16:9042 datacenter1>, <Host: 127.101.142.13:9042 datacenter1>, <Host: 127.101.142.19:9042 datacenter1>]

    async def check_auth_v2_works(manager: ManagerClient, hosts):
        cql = manager.get_cql()
        roles = set()
        for row in await cql.run_async("LIST ROLES"):
            roles.add(row.role)
        assert roles == set(["cassandra", "user 1", "user 2", "users"])
    
        user1_roles = await cql.run_async("LIST ROLES OF 'user 1'")
        assert len(user1_roles) == 2
        assert set([user1_roles[0].role, user1_roles[1].role]) == set(["users",  "user 1"])
    
>       await cql.run_async("CREATE ROLE user_after_migration")
E       cassandra.InvalidRequest: Error from server: code=2200 [Invalid query] message="Role user_after_migration already exists."

test/auth_cluster/test_auth_v2_migration.py:120: InvalidRequest
```

see https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/8086/artifact/testlog/x86_64/debug/auth_cluster.test_auth_v2_migration.3.log

**Comment by tchaikov** [comment]
[auth_cluster.test_auth_v2_migration.3.log](https://github.com/scylladb/scylladb/files/15046382/auth_cluster.test_auth_v2_migration.3.log)
[debug.zip](https://github.com/scylladb/scylladb/files/15046386/debug.zip)

**Comment by tchaikov** [comment]
hi @nuivall hi Marcin, do you mind taking a look?

**Comment by nuivall** [comment]
@tchaikov we never run this test twice on given `workdir` right? I see some raft concurrent modification may be involved here. I'm suspecting some incorrect behavior of the driver (this https://github.com/scylladb/python-driver/issues/296 which I couldn't reproduce).

So perhaps:
1. driver sent 2 create role requests (to different nodes)
2. one succeeded
3. another ran into concurrent modification exception, was retried internally and failed as the first succeeded already

I can't reproduce this locally and I think we don't have any logs wrt point 2. 

I plan to enable some query processor logs in test.py, just need to check if it's not logging to much.

**Comment by nuivall** [comment]
Since https://github.com/scylladb/scylladb/pull/18413 was merged I'm waiting for next occurrence with extra logs.

**Comment by yaronkaikov** [comment]
Seen again in https://jenkins.scylladb.com/job/scylla-master/job/next/7565/

**Comment by bhalevy** [comment]
Seen again in https://jenkins.scylladb.com/view/master/job/scylla-master/job/next/7583/artifact/testlog/x86_64/debug/auth_cluster.test_auth_v2_migration.3.log

**Comment by ptrsmrn** [comment]
Closing per https://github.com/scylladb/scylladb/issues/18931#issuecomment-2210575434

**Comment by lkshminarayanan** [comment]
Seen again in https://jenkins.scylladb.com/job/scylla-6.0/job/scylla-ci/148/testReport/junit/(root)/test_auth_v2_migration/Tests___Unit_Tests___test_auth_v2_migration_2/

**Comment by denesb** [comment]
> Seen again in https://jenkins.scylladb.com/job/scylla-6.0/job/scylla-ci/148/testReport/junit/(root)/test_auth_v2_migration/Tests___Unit_Tests___test_auth_v2_migration_2/

This is a 6.0 backport PR, can you please check if the fix was backported to 6.0? If not, simply backporting the fix will solve this on 6.0 too. In that case the issue doesn't have to be re-opened.

**Comment by lkshminarayanan** [comment]
> > Seen again in https://jenkins.scylladb.com/job/scylla-6.0/job/scylla-ci/148/testReport/junit/(root)/test_auth_v2_migration/Tests___Unit_Tests___test_auth_v2_migration_2/
> 
> This is a 6.0 backport PR, can you please check if the fix was backported to 6.0? If not, simply backporting the fix will solve this on 6.0 too. In that case the issue doesn't have to be re-opened.

PR https://github.com/scylladb/scylladb/pull/19484 adds more information to the test and is not a fix. It is also not backported and, this new failure in 6.0 doesn't provide any new information (https://github.com/scylladb/scylladb/issues/18931#issuecomment-2210575434). So closing it for now.
